android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt

```diff
@@ -139,8 +139,9 @@ internal class LocationUpdatesService : Service() {
     fun startTracking() {
         Logger.debug(TAG, "Requesting location updates")
         SharedPrefsUtil.saveIsTracking(this, true)
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && ActivityCounter.isAppInBackground()) {
             startForegroundService(Intent(applicationContext, LocationUpdatesService::class.java))
+            NotificationUtil.startForeground(this, location)
```

Comment:
This may have to be called in onCreate so perhaps an intent extra is needed

Reply:
In onStartCommand we can call stopSelf. If stopself is called before this startforeground fires, this will trigger a crash

Reply:
Same for stopTracking